src/components/NewMessageForm/NewMessageFormTextCreateDialog.vue

```diff
@@ -0,0 +1,266 @@
+<!--
+  - @copyright Copyright (c) 2022, Marco Ambrosini <marcoambrosini@icloud.com>
+  -
+  - @author Marco Ambrosini <marcoambrosini@icloud.com>
+  - @author Maksim Sukharev <antreesy.web@gmail.com>
+  -
+  - @license AGPL-3.0-or-later
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+  -->
+
+<template>
+	<NcModal size="large"
+		:container="container"
+		class="templates-picker"
+		@close="closeModal">
+		<div class="new-text-file">
+			<h2>
+				{{ t('spreed', 'Create and share a new file') }}
+			</h2>
+			<form class="new-text-file__form templates-picker__form"
+				:style="style"
+				@submit.prevent="handleCreateTextFile">
+				<NcTextField id="new-file-form-name"
+					ref="textField"
+					:error="!!newFileError"
+					:helper-text="newFileError"
+					:label="t('spreed', 'Name of the new file')"
+					:placeholder="textFileTitle"
+					:value="textFileTitle"
+					@update:value="updateTextFileTitle" />
+
+				<template v-if="fileTemplate.templates.length">
+					<ul class="templates-picker__list">
+						<NewMessageFormTemplatePreview v-bind="emptyTemplate"
+							:checked="checked === emptyTemplate.fileid"
+							@check="onCheck" />
+
+						<NewMessageFormTemplatePreview v-for="template in fileTemplate.templates"
+							:key="template.fileid"
+							v-bind="template"
+							:checked="checked === template.fileid"
+							:ratio="fileTemplate.ratio"
+							@check="onCheck" />
+					</ul>
+				</template>
+
+				<div class="new-text-file__buttons">
+					<NcButton type="primary"
+						@click="handleCreateTextFile">
+						{{ t('spreed', 'Create file') }}
+					</NcButton>
+				</div>
+			</form>
+		</div>
+	</NcModal>
+</template>
+
+<script>
+import { showError } from '@nextcloud/dialogs'
+
+import NcButton from '@nextcloud/vue/dist/Components/NcButton.js'
+import NcModal from '@nextcloud/vue/dist/Components/NcModal.js'
+import NcTextField from '@nextcloud/vue/dist/Components/NcTextField.js'
+
+import NewMessageFormTemplatePreview from './NewMessageFormTemplatePreview.vue'
+
+import { useViewer } from '../../composables/useViewer.js'
+import { createTextFile, shareFile } from '../../services/filesSharingServices.js'
+
+export default {
+	name: 'NewMessageFormTextCreateDialog',
```

Comment:
Not sure about the name "TextCreate". For general "text" creation we use the main input field. But this dialog is for creating a file.

I'd call something like:
- `NewTextFileDialog`
- `TextFileDialog`
- `ShareNewTextFileDialog`

Also, it seems that it is a pure component, not coupled with the new message form, except for opening the viewer. We may (but do not have to) make it a shared component, not `NewMessageForm_Part`.

Reply:
As long as it is used only in one place, I'd like to keep the name
Also, this is used not for .md files, but also for other extensions, so renamed it into `NewMessageNewFileDialog`